Alert glitches fixes

SychO9 opened the pull request:
Each commit has an explanation of the fix.

SychO9 committed:
Fixes #5518

Signed-off-by: SychO <sychocouldy@gmail.com>

SychO9 committed:
Fix dynamically received alerts

Additionally it makes it so that when new alerts are received and the popup has already been opened, it resets the popup to load the new alerts instead

Signed-off-by: SychO <sychocouldy@gmail.com>

SychO9 committed:
Fix a glitch where new alerts count never goes up unless the popup is opened

updateMemberData() has to be executed **after** the new alert(s) have been inserted, because the function itself executes alert_count() to assing to the member's "alerts" column, so before it didn't count the new alert(s) which lead to glitches

Singed-off-by: SychO <sychocouldy@gmail.com>

SychO9 commented on Themes/default/scripts/alerts.js:
```diff
@@ -51,7 +51,7 @@ var updateAlerts = function ()
 			var notification = notify.createNotification(new_alert_title, {
-				body: $(item).find('div.details:first > span').text(),
+				body: $(item).find('div.details:first span.alert_text').text(),
 				icon: smf_images_url + '/blank.png'
```
Also, shouldn't we allow admin to choose an icon for the desktop notifications instead of always a blank image ?

Sesquipedalian replied:
I'd say just use the OpenGraph image, falling back to the favicon.

SychO9 committed:
Use OpenGraph image for desktop notifications

Signed-off-by: SychO <sychocouldy@gmail.com>

SychO9 committed:
Close the desktop notification after opening it

Signed-off-by: SychO <sychocouldy@gmail.com>

SychO9 committed:
Fix frequency 2 setting not working

Turning "Straight away (but only for the first unread message)" frequency notification setting doesn't work because here $frequence is a string and FREQUENCY_FIRST_UNREAD_MSG is an integer so "2" !== 2 bu "2" == 2

Signed-off-by: SychO <sychocouldy@gmail.com>

MissAllSunday commented on Themes/default/Profile.template.php:
```diff
@@ -125,6 +125,8 @@ function(data) {
 						$("#alerts_menu_top span.amt").remove();
 						$("#alerts_menu div.alerts_unread").html(data);
+						if (typeof localStorage != "undefined")
+							localStorage.setItem("alertsCounter", 0);
```
What will happen if the browser doesn't support localStorage?

SychO9 replied:
If the browser doesn't support localStorage then there'll be no need to update the localStorage's alertsCounter value in the first place, because there'd be none

https://github.com/SychO9/SMF2.1/blob/ebbdad9f3806b1dc86fd912f9006dc0d5ee26478/Themes/default/scripts/alerts.js#L54-L55
